module/zstd/zstd.c

```diff
@@ -268,7 +263,36 @@ zstd_mempool_free(struct zstd_kmem *z)
 	mutex_exit(&pool->barrier);
 }
 
+struct levelmap {
+	int32_t cookie;
+	enum zio_zstd_levels level;
+};
 
+static struct levelmap fastlevels[] = {
+	{-1, ZIO_ZSTDLVL_FAST_1},
+	{-2, ZIO_ZSTDLVL_FAST_2},
+	{-3, ZIO_ZSTDLVL_FAST_3},
+	{-4, ZIO_ZSTDLVL_FAST_4},
+	{-5, ZIO_ZSTDLVL_FAST_5},
+	{-6, ZIO_ZSTDLVL_FAST_6},
+	{-7, ZIO_ZSTDLVL_FAST_7},
+	{-8, ZIO_ZSTDLVL_FAST_8},
+	{-9, ZIO_ZSTDLVL_FAST_9},
+	{-10, ZIO_ZSTDLVL_FAST_10},
+	{-20, ZIO_ZSTDLVL_FAST_20},
+	{-30, ZIO_ZSTDLVL_FAST_30},
+	{-40, ZIO_ZSTDLVL_FAST_40},
+	{-50, ZIO_ZSTDLVL_FAST_50},
+	{-60, ZIO_ZSTDLVL_FAST_60},
+	{-70, ZIO_ZSTDLVL_FAST_70},
+	{-80, ZIO_ZSTDLVL_FAST_80},
+	{-90, ZIO_ZSTDLVL_FAST_90},
+	{-100, ZIO_ZSTDLVL_FAST_100},
+	{-500, ZIO_ZSTDLVL_FAST_500},
+	{-1000, ZIO_ZSTDLVL_FAST_1000},
+};
+
+#define	ARRAYSIZE(array) sizeof (array) / sizeof (array[0])
```

Comment:
what about using `ARRAY_SIZE`?

Reply:
i was looking in zfs for such a macro. linux has it. but zfs must deal with bsd as well. i cannot straight use linux only macros in that code

Reply:
you should have looked at all other code... ARRAY_SIZE is used all over zfs^^
see `lib/libspl/include/os/linux/sys/sysmacros.h` how the linux "problem" is solved

Reply:
@c0d3z3r0 I agree, in some cases (such as Allan's code) it might be worth while keeping BSD compatibility fixes if they are already in. In some cases (such as here) we rather not already go creating workarounds and let the port team handle it.

Reply:
i was looking for it, but did not found it in header. will do that later this evening after i'm back at home

Reply:
fixed